src/components/DuetRoom.tsx

```diff
@@ -145,16 +167,16 @@ const DuetRoom: React.FC<{ maybeRoomId: string | null; isCreate: boolean }> = ({
       : undefined;
 
   // Piece information
-  const piece = JSON.parse(SamplePiece);
-  const notes: Array<Note> = piece.notes; // TODO: get the right notes
+  const pieceMIDI = JSON.parse(chosenSongMIDI);
```

Comment:
Should we memoize this? It might be expensive to parse the JSON for every re-render.

Reply:
Alternatively, pass in the parsed song from outside?

Reply:
Or should it just store the JSON in the state, instead of the string?

Reply:
That works too